backtracepython/utils/__init__.py

```diff
@@ -0,0 +1,16 @@
+import json
+import logging
+import re
+
+def parse_json(data, log_warning=True):
+    """Parses JSON from string. This uses the standard library json.dumps, but also adds
+    support for parsing illegal JSON values as null.
+    When illegal JSON characters (Nan, -Infinity, and Infinity) a warning log is emitted.
+    """
+    try:
+        return json.dumps(data, allow_nan=False)
+    except:
+        # warn user that they are passing invalid data. Parse with illegal JSON and regex replace with `null`.
+        if log_warning:
+            logging.warning('Illegal JSON values (NaN, -Infinity, and Infinity) will be parsed as nulls. Avoid passing these values in attributes. Read more here: https://docs.python.org/3/library/json.html')
+        return re.sub(r': (NaN|Infinity|-Infinity),?\b', ': null', json.dumps(data, allow_nan=True))
```

Comment:
This is probably best effort and acceptable, but will this not break for the below?

```py
        data_evil = {
            ": NaN": "NaN",
            ": Infinity": "Infinity",
            ": -Infinity": "-Infinity",
            "innocuous": "Totally a string: Infinity",
        }
```

PS: is there a reason why this is not shortened to `: (NaN|-?Infinity),?\b`?

Reply:
I agree with Pawel. Seeing this is exceptional before I think the deep clone / scrub NaN/Infinity starts making more sense.

Reply:
Yes, however it will only replace the keys if they have the format.

```
        data_evil = {
            ": NaN,": "NaN",
            ": Infinity,": "Infinity",
            ": -Infinity,": "-Infinity",
            "innocuous": "Totally a string: Infinity,",
        }
```
Note the `,` at the end of the key `": NaN,"` as a key is quite an edge case.Tthe real downside in the regex is the replacing of `"Totally a string: Infinity,",` cases - which is why the log warning is necessary as it can mess up data, and we are not able to guarantee valid JSON in some unknown edge cases we may not be thinking of.

That said, I think the alternative to this is to just use [simplejson](https://pypi.org/project/simplejson/) as a dependency and have it as an `install_requires` in setup.py, since it correctly will handle these and properly convert to `NaN` to `null`.

Should I continue with that?